### PR TITLE
Readme

### PR DESCRIPTION
**Pull request by bcoca**
add readme's to scenarios

### CODE REVIEW

File: migrate.py [hunk 1]
```diff
@@ -1652,6 +1652,9 @@ def main():
     spec = {}
 
     for spec_file in os.listdir(args.spec_dir):
+        if not spec_file.endswith('.yml'):
+            logger.debug('skipping %s as it is not a yaml file' % spec_file)
```
Comment (webknjaz) [inline migrate.py]: "yaml" may confuse users with `*.yaml`
```suggestion
            logger.debug('skipping %s as it is not a `*.yml` file' % spec_file)
```

File: migrate.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+                - module_utils: the most obvious one, you go from ``import ansible.module_utils.randomdir.myutil`` to ``import ansible_collections.myname.mycoll.randomdir.myutil``.
+                - subclassing plugins: you need to follwo the same rules in changing paths and using namespaced names.
+
+        - patches and mocks: mostly used in tests, these also might need rewrites with the same rules as imports follow.
+
+        - sanity tests: ignores need to be moved to ansible specific version ignores files ignores-2.9.txt
+
+        - integration tests and roles:
+                - tasks need to be rewritten to use FQCN, this includes modules, lookups, filters and any other plugin referenced that has been moved to a collection.
+				- Jinja filters and tests FQCN includes the file name where the plugin resides.
```
Comment (webknjaz) [inline migrate.md]: Broken indent
```suggestion
                - Jinja filters and tests FQCN includes the file name where the plugin resides.
```

File: migrate.md [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+repository: null # optionally repo where this collection lives
+documentation: null
+homepage: null
+issues: null
+```
+
+ .github/BOTMETA.yml'
+====================
+
+This file is used by the core issue management bot and other tools to keep track of many things, we added a ``migrated_to`` field that will indicate the collection name of where a file is supposed to primarily reside now.
```
Comment (webknjaz) [inline migrate.md]: double backticks is RST, not MD
```suggestion
This file is used by the core issue management bot and other tools to keep track of many things, we added a `migrated_to` field that will indicate the collection name of where a file is supposed to primarily reside now.
```

File: migrate.md [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+
+This project is using migrate.py to do automated migration from ansible/ansible to separate repos that can host 1 or more collections per repo, but there are those that might want to do a manual migration themselves.
+
+So how do you move existing code into a collection? Below we describe the steps that migrate.py does automatically but might not align with the structure you might want for your content.
+
+
+galaxy.yml
+==========
+
+The first thing is creating the file that 'defines' your collection, this shoudl have the namespace and collection name as well as any other collections you depend on for content.
```
Comment (mkrizek) [inline migrate.md]:
```suggestion
The first thing is creating the file that 'defines' your collection, this should have the namespace and collection name as well as any other collections you depend on for content.
```

File: migrate.md [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+license: GPLv3+
+tags: [stuff, morestuff, mystuff]
+dependencies: null # collections i depend on
+repository: null # optionally repo where this collection lives
+documentation: null
+homepage: null
+issues: null
+```
+
+ .github/BOTMETA.yml'
```
Comment (mkrizek) [inline migrate.md]:
```suggestion
 .github/BOTMETA.yml
```

File: migrate.md [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+You only need to update this file if your content is currently in ansible/ansible and can be ignored for 3rd party distributed content.
+
+migrating plugins
+=================
+
+The modules and the other plugin types require some rewrites to support collection semantics, mostly do deal with how they reference other resources, mostly module_utils and doc_fragments.
+
+
+ - paths:
+        - The first ting to note, all runtime code is under the ``plugins/`` path, so modules, module_utils and all otherplugins are in their own plugin specific directory under the plugins/ top level directory in the collection.
```
Comment (mkrizek) [inline migrate.md]:
```suggestion
        - The first thing to note, all runtime code is under the ``plugins/`` path, so modules, module_utils and all other plugins are in their own plugin specific directory under the plugins/ top level directory in the collection.
```

File: migrate.md [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+          This is similar to what already exists in roles, but for modules and module_utils it is a change from the core repo.
+        - tests now live under the ``tests/`` directory with ``units`` and ``integration`` subdirectories.
+
+ - Collection roles exist under the roles/ directory, with a couple of restrictions:
+        - They cannot have their own plugins, so  plugins in role adjacent directories will be ignored
+        - Role 'dependencies' will not be installed on collection install, if using collections those dependencies should be specified in galaxy.yml
+
+ - for most plugins you need to rewrite:
+        - documentation:
```
Comment (mkrizek) [inline migrate.md]: We also handle `version_added` (removing it currently).

File: migrate.md [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+                - doc_fragments: must use FQCN, examine all modules and plugins that use doc_fragments and rewrite from 'myfrag' to 'myname.mycoll.myfrag'.
+                - general docs: any references to modules and other plugins need to be rewritten to use FQCN, normally found in descriptions and examples.
+
+        - imports (also for tests):
+                - relative imports work in collections, but only if they start with a ``.`` so ``./filename`` and ``../asdfu/filestuff`` works but ``filename`` in same dir does not.
+                - module_utils: the most obvious one, you go from ``import ansible.module_utils.randomdir.myutil`` to ``import ansible_collections.myname.mycoll.randomdir.myutil``.
+                - subclassing plugins: you need to follwo the same rules in changing paths and using namespaced names.
```
Comment (mkrizek) [inline migrate.md]:
```suggestion
                - subclassing plugins: you need to follow the same rules in changing paths and using namespaced names.
```

File: migrate.md [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+        - patches and mocks: mostly used in tests, these also might need rewrites with the same rules as imports follow.
+
+        - sanity tests: ignores need to be moved to ansible specific version ignores files ignores-2.9.txt
+
+        - integration tests and roles:
+                - tasks need to be rewritten to use FQCN, this includes modules, lookups, filters and any other plugin referenced that has been moved to a collection.
+				- Jinja filters and tests FQCN includes the file name where the plugin resides.
+
+
+There are many corner cases if you want to automat the points above, see the `migrate.py` script for specifics.
```
Comment (mkrizek) [inline migrate.md]:
```suggestion
There are many corner cases if you want to automate the points above, see the `migrate.py` script for specifics.
```

File: scenarios/newworld/README.md [hunk 10]
```diff
@@ -0,0 +1 @@
+Translation from a prposed spreadsheet, not really in use.
```
Comment (mkrizek) [inline scenarios/newworld/README.md]:
```suggestion
Translation from a proposed spreadsheet, not really in use.
```

File: scenarios/minimal/README.md [hunk 11]
```diff
@@ -0,0 +1 @@
+Opinionated 'minimal' usable ansbile version, keeps certain things in core that are either required for any ansible cli to execute or allow minimal configuration of the system to then enable installing collections and/or other content.
```
Comment (mkrizek) [inline scenarios/minimal/README.md]:
```suggestion
Opinionated 'minimal' usable ansible version, keeps certain things in core that are either required for any ansible cli to execute or allow minimal configuration of the system to then enable installing collections and/or other content.
```

File: migrate.md (hunk 8 above)
Comment (mkrizek) [reply migrate.md]: Do we actually handle subclassing plugins?
